collector/aws/elasticips.go

```diff
@@ -0,0 +1,186 @@
+	Metric        string
+	IP            string
+	PricePerHour  float64
+	PricePerMonth float64
+	Tag           map[string]string
+}
+
+// NewElasticIPManager implements AWS GO SDK
+func NewElasticIPManager(collector collector.CollectorDescriber, client ElasticIPClientDescreptor, pricing *PricingManager, metric config.ResourceConfig, region string) *ElasticIPManager {
+
+	return &ElasticIPManager{
+		collector:          collector,
+		client:             client,
+		pricingClient:      pricing,
+		metric:             metric,
+		region:             region,
+		servicePricingCode: "AmazonEC2",
+		rateCode:           "JTU8TKNAMW",
+		Name:               fmt.Sprintf("%s_elastic_ip", ResourcePrefix),
+	}
+}
+
+// Detect checks if elastic ips is under utilize
+func (ei *ElasticIPManager) Detect() ([]DetectedElasticIP, error) {
+
+	log.WithFields(log.Fields{
+		"region":   ei.region,
+		"resource": "elastic ips",
+	}).Info("starting to analyze resource")
+
+	ei.collector.UpdateServiceStatus(collector.EventCollector{
+		ResourceName: ei.Name,
+		Data: collector.EventStatusData{
+			Status: collector.EventFetch,
+		},
+	})
+
+	elasticIPs := []DetectedElasticIP{}
+
+	priceFIlters := ei.GetPricingFilterInput()
+	// Getting elastic ip pricing
+	price, err := ei.pricingClient.GetPrice(priceFIlters, ei.rateCode, ei.region)
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"rate_code":     ei.rateCode,
+			"region":        ei.region,
+			"price_filters": priceFIlters,
+		}).Error("could not get elastic ip price")
+		ei.updateErrorServiceStatus(err)
+		return elasticIPs, err
+	}
+
+	// Getting all elastic ip addressess
+	ips, err := ei.DescribeAddressess()
+	if err != nil {
+		ei.updateErrorServiceStatus(err)
+		return elasticIPs, err
+	}
+
+	for _, ip := range ips {
+
+		// Checks id the ip that are not associated with a running Amazon Elastic Compute Cloud (Amazon EC2) instance
+		if ip.PrivateIpAddress == nil && ip.AssociationId == nil && ip.InstanceId == nil && ip.NetworkInterfaceId == nil {
+
+			tagsData := map[string]string{}
+			if err == nil {
+				for _, tag := range ip.Tags {
+					tagsData[*tag.Key] = *tag.Value
+				}
+			}
+
+			eIP := DetectedElasticIP{
+				Region:        ei.region,
+				Metric:        ei.metric.Description,
+				IP:            *ip.PublicIp,
+				PricePerHour:  price,
+				PricePerMonth: price * collector.TotalMonthHours,
+				Tag:           tagsData,
+			}
+
+			ei.collector.AddResource(collector.EventCollector{
+				ResourceName: ei.Name,
+				Data:         eIP,
+			})
+
+			elasticIPs = append(elasticIPs, eIP)
+
+		}
+	}
+
+	ei.collector.UpdateServiceStatus(collector.EventCollector{
+		ResourceName: ei.Name,
+		Data: collector.EventStatusData{
+			Status: collector.EventFinish,
+		},
+	})
+
+	return elasticIPs, nil
+
+}
+
+// GetPricingFilterInput return the elastic ip price filters.
+func (ei *ElasticIPManager) GetPricingFilterInput() *pricing.GetProductsInput {
+
+	input := &pricing.GetProductsInput{
```

Comment:
Why do we need this variable "input" ? 
Why not return the struct.

Reply:
changed